wp_api/src/request.rs

```diff
@@ -162,10 +166,23 @@ pub struct WpNetworkRequest {
 
 #[uniffi::export]
 impl WpNetworkRequest {
+    /// Unique identifier for this request – used for external record-keeping
     pub fn request_id(&self) -> String {
         self.uuid.clone()
     }
 
+    /// Increments the retry count for this request by returning a copy of it
+    pub fn increment_retry_count(&self) -> Self {
+        let mut request = self.clone();
+        request.retry_count += 1;
+        request
+    }
```

Comment:
The following suggestion avoids deriving `Clone` and improves(?) the name of the function. We could maybe call it `retry` as well, because it'd be clear what it does from the types, but this might be a bit more clear at a first glance.

I don't like the name `increment_retry_count`, because it'd be something that takes `&mut self` and increment it in place. Although similar to `retry` suggestion, it's clear from the function signature that it returns a copy of the original request.

```suggestion
    /// Clones the request and increments the retry count
    pub fn clone_with_incremented_retry_count(&self) -> Self {
        Self {
            uuid: self.uuid.clone(),
            retry_count: self.retry_count + 1,
            method: self.method.clone(),
            url: self.url.clone(),
            header_map: Arc::clone(&self.header_map),
            body: self.body.clone(),
        }
    }
```